### PR TITLE
SONARJAVA-1927 Expose 'isOverriding()' and 'overridenSymbol()' in API

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/plugins/java/api/tree/MethodTree.java [hunk 1]
```diff
@@ -90,4 +88,10 @@
 
   Symbol.MethodSymbol symbol();
 
+  /**
+   * Check if a methodTree is overriding any other method. The corresponding overridden symbol can be retrieved through the {@link #symbol()}.
+   *
+   * @return true if overriding, null if it cannot be decided (method symbol not resolved or lack of byte code for super types), false if not overriding.
+   */
+  Boolean isOverriding();
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/plugins/java/api/tree/MethodTree.java]: isn't this lacking an annotation  ?

Comment (m-g-sonar) [reply java-frontend/src/main/java/org/sonar/plugins/java/api/tree/MethodTree.java]: Good catch! I forgot the `@CheckForNull` annotation!

File: java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java [hunk 2]
```diff
@@ -269,9 +263,9 @@ public Boolean isOverriding() {
     if (symbol == null) {
       return null;
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: Code Smell Code Smell: Null is returned but a "Boolean" is expected. ([squid:S2447](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2447))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&branch=SONARJAVA-1927&issues=AWDq5lGyo_TTbi26M3Q9&open=AWDq5lGyo_TTbi26M3Q9)

[ISSUE_KEY]: AWDq5lGyo_TTbi26M3Q9

Comment (m-g-sonar) [reply java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: Looks like a FP to me. Method is annotated with `@Nullable`.

Comment (m-g-sonar) [reply java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: https://jira.sonarsource.com/browse/SONARJAVA-2616

Comment (SonarTech) [reply java-frontend/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED

**Pull request merged**